USSF-ORBIT/ussf-portal-client
# feat: Add event tracking to existing features

## Description 

This PR converts the `useAnalytics` hook to a React context, to ensure the analytics script is only initialized once on the window, and the `push` method can be used across components without having to drill props down. I also added a `trackEvent` method that is just a shortcut for `push(['trackEvent'...`.

I added `trackEvent` to all user events in the [spec](https://docs.google.com/spreadsheets/d/1iQJOdfqQVk_WxmBmg0DoIPuc0EL_BwxtpT5MyoeJ8jE/edit#gid=0) that aren't being modified in other PRs.

Fixes #471 

## Review Notes

To test locally:
* Start up the app and Matomo using docker services
    *  `yarn build && yarn start`
    *  `docker-compose -f docker-compose.services.yml up -d`
* Log into the app and click through some of the actions that should fire events
* Verify that you see the events being tracked by Matomo 
    *  Matomo: http://localhost/
    * Log in: `User / bitnami`
